src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/Utf8BufferTextReader.cs

```diff
@@ -11,10 +11,12 @@ namespace Microsoft.AspNetCore.SignalR.Internal.Protocol
     internal class Utf8BufferTextReader : TextReader
     {
         private ReadOnlyMemory<byte> _utf8Buffer;
+        private Decoder _decoder;
 
         public Utf8BufferTextReader(ReadOnlyMemory<byte> utf8Buffer)
         {
             _utf8Buffer = utf8Buffer;
+            _decoder = Encoding.UTF8.GetDecoder();
```

Comment:
Can this be cached statically?

Reply:
It's stateful and resttable so sorta but not really. Also the things did slow down a bit but it is more correct (after writing some unicode tests).